src/nvim/api/win_config.c

```diff
@@ -639,6 +656,63 @@ void nvim_win_set_config(Window window, Dict(win_config) *config, Error *err)
       win_setheight_win(fconfig.height, win);
     }
   } else {
+    if (win->w_floating && parent && parent_tp != win_tp) {
+      if (!can_move(win, win_tp != parent_tp, err)) {
+        return;
+      }
+      win_T *altwin = NULL;
+
+      if (win == curwin) {
+        altwin = win_float_find_altwin(win, NULL);
+        win_goto(altwin);
+
+        if (curwin == win) {
+          api_set_error(err, kErrorTypeException, "Failed to switch away from window %d",
+                        win->handle);
+          return;
+        }
+
+        if (!win_valid_any_tab(win)) {
+          api_set_error(err, kErrorTypeException, "Window was closed by autocommands %d",
+                        win->handle);
+          return;
+        }
+
+        win_tp = win_find_tabpage(win);
+        parent_tp = win_find_tabpage(parent);
+      }
+
+      if (win_tp != parent_tp) {
+        win_remove(win, win_tp == curtab ? NULL : win_tp);
+        win_T *target_after;
+        if (parent_tp == curtab) {
+          target_after = lastwin_nofloating();
+        } else {
+          target_after = parent_tp->tp_lastwin;
+          while (target_after && target_after->w_floating) {
+            target_after = target_after->w_prev;
+          }
+          if (target_after == NULL) {
+            target_after = parent_tp->tp_lastwin;
+          }
+        }
+
+        win_append(target_after, win, parent_tp == curtab ? NULL : parent_tp);
+        if (win_tp->tp_curwin == win) {
+          win_tp->tp_curwin = altwin ? altwin : win_float_find_altwin(win, win_tp);
+        }
```

Comment:
```suggestion
        // If `win` was the curwin of its old tabpage, select a new curwin for it.
        if (win_tp->tp_curwin == win) {
          win_tp->tp_curwin = win_float_find_altwin(win, win_tp);
        }
```

Don't reuse `altwin`; `win_goto` autocommands may have freed it.